# E41 Wave 4 - the Epoch-Matched Cohort Reference

**Author**: kj · **Round**: E41 Wave 4 (research wave, no new hypotheses)

The E41 Stage-5 honesty clause failed as pre-registered - 3/6 non-exempt regions' honesty gap
D = |PB0 − PB0_check|/PB0_check WORSENED under the data-derived anchors - and the failure was
attributed, not excused: the b.1975 completed-cohort reference is a pre-collapse world, so D
mixes calibration honesty with genuine period-vs-cohort EROSION. This wave executes the named
fix: rebuild the check on a **b.1985 spliced pseudo-cohort** (the observed ASFR diagonal ages
15-38, years 2000-2023, completed above age 38 with the 2023 period schedule) evaluated
against the SHIPPED state pair - an epoch-matched reference in both fertility and composition,
so the residual D reads calibration alone and the b.1975-vs-b.1985 difference MEASURES the
erosion the old FAIL only asserted.

**Pre-registered success bars** (a diagnostics refresh, adjudicated but not hypothesis-graded):
(i) >= 5/6 non-exempt regions have D_epoch < D_ref (the reference, not the calibration, was
the drift); (ii) Korea's D_epoch < 0.15 (its state pair is period-epoch by construction, so
the epoch-matched check must nearly close); (iii) every splice-completion share < 15% of the
CFR (the borrowed 2023 tail stays a minor correction).

**Output**: `reports/e41_w4_cohort_refresh.json`; records - the experiments-log E41 addendum,
the SOTA "cohort reference is receding" limitation updated.

## Imports

In [1]:
%load_ext autoreload
%autoreload 2

import os
os.environ.setdefault("CUDA_VISIBLE_DEVICES", "GPU-58ae1f45-295c-681b-60ad-843265f52997")
import sys
sys.path.insert(0, "../src")

import json
from pathlib import Path

import numpy as np
from rich import print as rprint

from sci_demographic_collapse import emergent as em
from sci_demographic_collapse.emergent import C0, REAL, RV0, fec

2026-07-11 16:27:43.951 | INFO     | sci_demographic_collapse.config:<module>:40 - PROJ_ROOT path is: /home/lab/workspace/learning/projects/sci-demographic-collapse


## Reproducibility

In [2]:
SEED = 0   # fully deterministic - observed data + closed-form checks, no simulation

## The spliced b.1985 pseudo-cohort

For each region, the cohort born 1985 is observed from age 15 (year 2000) to age 38 (2023) in
the on-disk WPP ASFR array; ages 39-48 are completed with the 2023 period schedule (the
freeze-rate splice, same construction as NB37's Stage 3b but one decade later). The spliced
CFR and its schedule-weighted mean age give the epoch-matched fertility reference.

In [3]:
m = em.EmergentModel("../data/raw/unwpp")
REGIONS = list(REAL)
BIRTH = 1985
SPL = {}
for r in REGIONS:
    A = m.REG[r]["asfr"]          # rows = years 1990-2023, cols = ages
    obs_ages = list(range(15, 39))                     # observed: years 2000..2023
    tail_ages = list(range(39, 49))                    # completed with the 2023 schedule
    f_obs = np.array([A[BIRTH + a - 1990][a] for a in obs_ages])
    f_tail = np.array([A[2023 - 1990][a] for a in tail_ages])
    ages = np.array(obs_ages + tail_ages, dtype=float)
    f = np.concatenate([f_obs, f_tail])
    cfr = float(f.sum())
    mac = float((ages * f).sum() / max(f.sum(), 1e-9))
    SPL[r] = dict(CFR_spl=cfr, MAC_spl=mac, tail_share=float(f_tail.sum() / max(cfr, 1e-9)))
    rprint(f"  {r:8s} CFR(b.{BIRTH} spliced) {cfr:.3f}   MAC {mac:.2f}   completion share {SPL[r]['tail_share']:.1%}")

USA      CFR(b.1985 spliced) 2.083   MAC 27.75   completion share 4.7%

France   CFR(b.1985 spliced) 2.067   MAC 29.81   completion share 6.8%

Germany  CFR(b.1985 spliced) 1.614   MAC 30.22   completion share 6.7%

Italy    CFR(b.1985 spliced) 1.410   MAC 30.83   completion share 8.8%

Japan    CFR(b.1985 spliced) 1.476   MAC 30.36   completion share 6.5%

Korea    CFR(b.1985 spliced) 1.209   MAC 30.92   completion share 5.3%

Poland   CFR(b.1985 spliced) 1.442   MAC 28.59   completion share 4.2%

Israel   CFR(b.1985 spliced) 3.090   MAC 29.88   completion share 7.3%

## The epoch-matched honesty check

`PB0_check_epoch = CFR_spliced / (C0 · (1−RV0) · fec(MAC_spliced))` with the SHIPPED state
pair - fertility and composition from the same post-2010 epoch. D_epoch = |PB0 −
check|/check then reads calibration honesty alone; D_ref − D_epoch is the measured erosion
the Stage-5 FAIL attributed but never quantified.

In [4]:
S5 = json.load(open("../reports/e41_stage5_reverdict.json"))["honesty_table"]
W4 = {}
for r in REGIONS:
    chk = SPL[r]["CFR_spl"] / (C0[r] * (1 - RV0[r]) * fec(SPL[r]["MAC_spl"]))
    pb0 = float(m.PB0[r])
    d_e = abs(pb0 - chk) / chk
    d_ref = float(S5[r]["D_new"])
    W4[r] = dict(PB0=pb0, check_epoch=float(chk), D_epoch=float(d_e), D_ref=d_ref,
                 erosion=float(d_ref - d_e), exempt=bool(S5[r]["exempt"]),
                 CFR_spl=SPL[r]["CFR_spl"], MAC_spl=SPL[r]["MAC_spl"],
                 tail_share=SPL[r]["tail_share"])
    tag = "EXEMPT" if W4[r]["exempt"] else ("better " if d_e < d_ref else "WORSE  ")
    rprint(f"  {r:8s} PB0 {pb0:.3f}  check_epoch {chk:.3f}  D_epoch {d_e:.3f} (was {d_ref:.3f})  "
           f"erosion {d_ref - d_e:+.3f}  {tag}")

nonex = [r for r in REGIONS if not W4[r]["exempt"]]
n_better = sum(W4[r]["D_epoch"] < W4[r]["D_ref"] for r in nonex)
korea_ok = W4["Korea"]["D_epoch"] < 0.15
tails_ok = all(W4[r]["tail_share"] < 0.15 for r in REGIONS)
n_within10 = sum(W4[r]["D_epoch"] < 0.10 for r in REGIONS)
rprint(f"\n  bar (i): {n_better}/{len(nonex)} non-exempt improve (bar >=5)  ->  {'PASS' if n_better >= 5 else 'FAIL'}")
rprint(f"  bar (ii): Korea D_epoch {W4['Korea']['D_epoch']:.3f} (bar <0.15)  ->  {'PASS' if korea_ok else 'FAIL'}")
rprint(f"  bar (iii): all completion shares <15%  ->  {'PASS' if tails_ok else 'FAIL'}")
rprint(f"  regions within the 10% honesty band on the epoch check: {n_within10}/8")
W4_OUTCOME = dict(bars=dict(i=bool(n_better >= 5), ii=bool(korea_ok), iii=bool(tails_ok)),
                  n_better=int(n_better), n_nonexempt=len(nonex), n_within10=int(n_within10))

USA      PB0 1.941  check_epoch 2.496  D_epoch 0.222 (was 0.718)  erosion +0.496  better

France   PB0 2.000  check_epoch 2.403  D_epoch 0.168 (was 0.361)  erosion +0.193  better

Germany  PB0 1.803  check_epoch 2.031  D_epoch 0.113 (was 0.172)  erosion +0.060  better

Italy    PB0 1.671  check_epoch 1.878  D_epoch 0.110 (was 0.235)  erosion +0.125  better

Japan    PB0 1.756  check_epoch 2.045  D_epoch 0.141 (was 0.209)  erosion +0.068  better

Korea    PB0 1.253  check_epoch 1.950  D_epoch 0.358 (was 0.631)  erosion +0.274  EXEMPT

Poland   PB0 1.402  check_epoch 1.743  D_epoch 0.196 (was 0.531)  erosion +0.336  better

Israel   PB0 3.152  check_epoch 3.353  D_epoch 0.060 (was 0.167)  erosion +0.107  EXEMPT

bar (i): 6/6 non-exempt improve (bar >=5)  ->  PASS

bar (ii): Korea D_epoch 0.358 (bar <0.15)  ->  FAIL

bar (iii): all completion shares <15%  ->  PASS

regions within the 10% honesty band on the epoch check: 1/8

## Save the record

In [5]:
out = dict(wave="E41-W4", birth_cohort=BIRTH,
           construction="observed ASFR diagonal ages 15-38 (years 2000-2023) + 2023-schedule tail 39-48; "
                        "check = CFR_spl / (C0*(1-RV0)*fec(MAC_spl)) on the SHIPPED state pair",
           table=W4, outcome=W4_OUTCOME,
           reference_superseded="b.1975 completed-cohort check (reports/e41_stage5_reverdict.json) - "
                                "kept as the erosion baseline, no longer the honesty gate")
Path("../reports").mkdir(exist_ok=True)
json.dump(out, open("../reports/e41_w4_cohort_refresh.json", "w"), indent=2)
rprint("[bold green]saved[/bold green] reports/e41_w4_cohort_refresh.json")

saved reports/e41_w4_cohort_refresh.json

## Conclusions

**The decontamination worked, and its one failure is itself the finding.**

- **Bar (i) PASS, 6/6**: every non-exempt region's honesty gap shrinks under the epoch-matched
  b.1985 reference (USA 0.718 -> 0.222, Poland 0.531 -> 0.196, France 0.361 -> 0.168, Italy
  0.235 -> 0.110, Japan 0.209 -> 0.141, Germany 0.172 -> 0.113) - the Stage-5 FAIL was the
  REFERENCE drifting, not the calibration, exactly as attributed. The per-region erosion is now
  measured, not asserted: USA +0.50 of D was pure epoch mismatch, Poland +0.34, France +0.19.
- **Bar (ii) FAIL, and honestly so**: Korea's epoch gap is 0.358 (down from 0.631 but far above
  the 0.15 bar). The b.1985 Korean cohort completed most of its fertility before the 2015-2023
  collapse floor, so even a decade-fresher completed cohort lags the period state - Korea's
  erosion OUTRUNS any completed-cohort reference, and only a circular period-on-period check
  could close it. This is the C0F2 erosion signal, now with a rate attached.
- **Bar (iii) PASS**: the borrowed 2023 tail is 4-9% of every spliced CFR - the splice is an
  observation, not a model echo.

The b.1975 table (reports/e41_stage5_reverdict.json) is retained as the erosion BASELINE; the
honesty gate going forward reads the epoch-matched check (reports/e41_w4_cohort_refresh.json).
Residual epoch gaps of 0.06-0.22 (ex-Korea) remain the honest statement of how far the period
calibration sits from even a current cohort's realized fertility.